File: leetcode/1598/crawlerLogFolder.py
class Solution:
    def minOperations(self, logs: List[str]) -> int:
        answer = 0
        for name in logs:
            if ".." in name and answer > 0:
                answer -=1
            elif "." in name:
                pass
            else:
                answer += 1
        return answer

File: leetcode/1598/crawlerLogFolder.swift
class Solution {
    func minOperations(_ logs: [String]) -> Int {
        var answer = 0
        var newname : String
        for name in logs{
            let idx = name.firstIndex(of:"/") ?? name.endIndex
            newname = String(name[..<idx])
            if ".." == newname {
                answer -= 1
                answer = max(answer, 0)
            }
            else if "."  == newname{
                continue
            }
            else{
                answer += 1
            }
        }
        return answer
    }
}
